src/check.rs

```diff
@@ -0,0 +1,88 @@
+use core::str;
+
+use esp_metadata::Chip;
+
+struct Version {
+    major: u8,
+    minor: u8,
+    patch: u8,
+}
+
+enum CheckResult {
+    Ok,
+    WrongVersion,
+    NotFound,
+}
+
+pub fn check(chip: Chip) {
+    let rust_version = get_version(
+        "cargo",
+        if chip.is_xtensa() {
+            &["+esp"]
+        } else {
+            &["+stable"]
+        },
+    );
+
+    let espflash_version = get_version("espflash", &[]);
+    let probers_version = get_version("probe-rs", &[]);
+
+    println!("\nChecking installed versions");
+    print_result("Rust", check_version(rust_version, 1, 84, 0));
+    print_result("espflash", check_version(espflash_version, 3, 3, 0));
+    print_result("probe-rs", check_version(probers_version, 0, 25, 0));
```

Comment:
Should we check both probe-rs and espflash regardless of the option the user selected?

Reply:
I thought about that before but it's just for the user's information - e.g. I usually use espflash for flashing but probe-rs for debugging - so IMHO it's not XOR and ideally at some point a user should have both (e.g. `espflash --erase-chip` is something occasionally needed even when using probe-rs for flashing)

Reply:
_but_ maybe showing a red cross here might make people feel uncomfortable - even if they won't need it - not sure

Reply:
We can give it a try and if we see users are confused about this, change it in the futrure